### PR TITLE
LLVM and Tests

### PR DESCRIPTION
**Pull request by sethitow**

### CODE REVIEW

**Review by OmoshiroiBoi — APPROVED** [review]
The changes look good, but I have some questions about a few of the different files that the changes are being made on.

- On CMakeLists.txt what is the original meaning for lines 11-16? Was it moved under src/CMakeLists.txt? 

- Was the file intake part removed out of parse.cpp to only be located in main.cpp?

- Can we briefly go over test/token.cpp again when you have time please?

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,19 +1,17 @@
...
 project(stc VERSION 0.1.0)
 
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED True)
 
 set(Boost_USE_STATIC_LIBS ON)
 find_package(Boost 1.70 REQUIRED COMPONENTS program_options)
+include_directories(${Boost_INCLUDE_DIRS}) 
 
-add_executable(stc src/main.cpp
```
Comment (OmoshiroiBoi) [inline CMakeLists.txt:11]: What is the difference between add_executable here and add_library down below?

File: src/parse.cpp [hunk 2]
```diff
@@ -1,28 +1,10 @@
...
-#include <fstream>
-#include <sstream>
-
 #include "lexer.hpp"
 
-int parse(std::string filename)
+// TODO: This return type might need to change
+std::unique_ptr<FunctionAST> parse(std::vector<Token> tokens)
 {
-
```
Comment (OmoshiroiBoi) [inline src/parse.cpp:10]: Are we leaving the file intake portion squarely in the main.cpp portion of the code?

File: test/parse_function.cpp [hunk 3]
```diff
@@ -0,0 +1,47 @@
+#include "gtest/gtest.h"
+
+#include "../src/ast/expr_ast.hpp"
+#include "../src/ast/function_ast.hpp"
+#include "../src/ast/prototype_ast.hpp"
+#include "../src/parse.hpp"
+#include "../src/token.hpp"
+
```
Comment (OmoshiroiBoi) [inline test/parse_function.cpp:8]: Should this also include

std::string input = "FUNCTION F_Sample : INT"
                        "  VAR_INPUT"
                        "    x : INT;"
                        "    y : INT;"
                        "  END_VAR"
                        "  F_Sample := x + y;"
                        "END_FUNCTION;";

**Review by OmoshiroiBoi — COMMENTED** [review]
Updated some comments

File: CMakeLists.txt (hunk 1 above)
Comment (sethitow) [reply CMakeLists.txt:11]: Add library creates a named library that can be linked against elsewhere, but doesn't generate a runnable binary. add_executable generates something you can actually run.

File: src/parse.cpp (hunk 2 above)
Comment (sethitow) [reply src/parse.cpp:10]: Correct, to better reflect the flow of lex->parse->generate, I moved the file reading code out of this function

File: test/parse_function.cpp (hunk 3 above)
Comment (sethitow) [reply test/parse_function.cpp:8]: No, this test suite only tests the parsing code, not the tokenizing code. Testing "units" of code independently helps isolate where the problem is. If lexer and parser were tested together, and the test failed, it wouldn't be obvious which portion of code had an issue.

**Pull request merged**